src/beta/resources/referral.js

```diff
@@ -0,0 +1,162 @@
+ */
+async function getEasyPostStripeKey(api) {
+  const response = await api.get('partners/stripe_public_key');
+
+  return response.body.public_key;
+}
+
+/**
+ * Send the credit card details to Stripe to get a Stripe credit card ID.
+ * @param {API} api - The API client to use.
+ * @param {string} creditCardNumber - Credit card number
+ * @param {string} creditCardExpirationMonth - Credit card expiration month
+ * @param {string} creditCardExpirationYear - Credit card expiration year
+ * @param {string} creditCardCVC - Credit card CVC
+ * @returns {Promise<string>} - Stripe credit card ID
+ */
+async function sendCardDetailsToStripe(
+  api,
+  creditCardNumber,
+  creditCardExpirationMonth,
+  creditCardExpirationYear,
+  creditCardCVC,
+) {
+  const stripeKey = await getEasyPostStripeKey(api);
+
+  // need to form-encode per Stripe's API
+  const request = superagent.post(stripeCreditCardUrl).set({
+    Authorization: `Bearer ${stripeKey}`,
+    'Content-Type': 'application/x-www-form-urlencoded',
+  });
+  request.send({
+    card: {
+      number: creditCardNumber,
+      exp_month: creditCardExpirationMonth,
+      exp_year: creditCardExpirationYear,
+      cvc: creditCardCVC,
+    },
+  });
+
+  try {
+    const response = await request;
+
+    return response.body.id;
+  } catch (error) {
+    if (error.response && error.response.body) {
+      throw new RequestError(error.response.body, stripeCreditCardUrl);
+    }
+    throw error;
+  }
+}
+
+/**
+ * Send the Stripe credit card ID to EasyPost to add the card to the user's account.
+ * @param {API} api - The API client to use.
+ * @param {string} referralApiKey - The referral user's production API key.
+ * @param {string} stripeCreditCardId - Stripe credit card ID
+ * @param {string} priority - Whether to add the card as the 'primary' or 'secondary' card
+ * @returns {Object} - Response body (EasyPost payment method object)
+ */
+async function sendCardDetailsToEasyPost(api, referralApiKey, stripeCreditCardId, priority) {
+  const _api = getReferralApi(api, referralApiKey);
+  const params = { credit_card: { stripe_object_id: stripeCreditCardId, priority } };
+  const response = await _api.post('credit_cards', params);
+
+  return response.body;
+}
+
+export default (api) =>
+  class Referral extends base(api) {
+    static _name = 'Referral';
+
+    static _url = 'referral_customers';
+
+    static key = 'user';
+
+    static propTypes = propTypes;
+
+    /**
+     * Update the referral's email address.
+     * @param {string} referralUserId - The referral user's ID.
+     * @param {string} email - The new email address.
+     * @returns {Promise<boolean>} - Returns true if the referral was updated successfully, false otherwise.
+     */
+    static async updateEmail(referralUserId, email) {
+      try {
+        const newParams = { user: { email } };
+        await api.put(`${this._url}/${referralUserId}`, newParams);
+
+        return true;
```

Comment:
We should be returning the message of the API, not a bool. This is how our delete methods across libs got all out of sync

Reply:
After discussing, I'm ok with returning `true` here as long as we try/catch in each lib so that we aren't blindly returning true (true will only get returned if no error).